classes/04class/aws_iam_storage/README.md

```diff
@@ -0,0 +1,540 @@
+# Introduction to Containers
+
+This class is an introduction to the AWS storage service S3, AWS IAM and database service RDS.
+
+The AWS CLI is also introduced as an import tool to help in the daily activities of a Cloud Engineer.
+
+**Contents**
+- [Introduction to Containers](#introduction-to-containers)
+- [AWS Command Line Interface (CLI)](#aws-command-line-interface-cli)
+  - [Why AWS CLI?](#why-aws-cli)
+  - [Installation (now CLIv2)](#installation-now-cliv2)
+  - [Configuration](#configuration)
+    - [Setting Access Keys and Secrets](#setting-access-keys-and-secrets)
+      - [Alternative method to set credentials (not recommended)](#alternative-method-to-set-credentials-not-recommended)
+      - [Checking who are you with AWS Security Token Service (STS)](#checking-who-are-you-with-aws-security-token-service-sts)
+  - [CLI Documentation and Help](#cli-documentation-and-help)
+  - [Example: List your S3 buckets](#example-list-your-s3-buckets)
+- [AWS Identity and Access Management (IAM)](#aws-identity-and-access-management-iam)
+  - [What's IAM?](#whats-iam)
+  - [IAM - Four Pillars (Users, Groups, Roles, Policies)](#iam---four-pillars-users-groups-roles-policies)
+    - [Users](#users)
+    - [Groups](#groups)
+    - [Roles](#roles)
+    - [Policies and Permissions](#policies-and-permissions)
+      - [Policy structure](#policy-structure)
+      - [Policy Evaluation Logic](#policy-evaluation-logic)
+  - [Let's not depend on users - Using STS to assume roles!](#lets-not-depend-on-users---using-sts-to-assume-roles)
+      - [A cleaner way to assume roles using AWS CLI](#a-cleaner-way-to-assume-roles-using-aws-cli)
+- [AWS Simple Storage Service (S3)](#aws-simple-storage-service-s3)
+  - [What's S3?](#whats-s3)
+  - [Main Use Cases](#main-use-cases)
+  - [Amazon S3 Concepts](#amazon-s3-concepts)
+    - [Buckets](#buckets)
+      - [Important stuff](#important-stuff)
+    - [Objects](#objects)
+      - [Important stuff](#important-stuff-1)
+    - [Keys](#keys)
+  - [Storage classes](#storage-classes)
+  - [When not to use it](#when-not-to-use-it)
+- [AWS Relational Database Service (RDS)](#aws-relational-database-service-rds)
+  - [Why RDS?](#why-rds)
+    - [Permissions](#permissions)
+    - [Database on EC2 vs RDS](#database-on-ec2-vs-rds)
+- [Main class takeaways](#main-class-takeaways)
+- [Appendix](#appendix)
+    - [Archiving/Backup - Amazon Glacier](#archivingbackup---amazon-glacier)
+
+# AWS Command Line Interface (CLI)
+
+## Why AWS CLI?
+The AWS Command Line Interface is a unified tool that provides a consistent interface for interacting with several AWS services.
+
+It is used by professionals on a daily basis to perform troubleshooting and automation, so it is an essential tool for any AWS professional.
+
+## Installation (now CLIv2)
```

Comment:
awscli was one of the tools that we have asked on the pre-class task. People should already have it installed and I would not recommend to have installation steps in our classes. Maybe just add a link to the file that we have the instructions about aws cli installation: https://github.com/devopsacademyau/academy/tree/master/classes/00pre-class#aws-command-line-tool
Maybe update that link to cli v2?

Reply:
I don't expect to go through it in class and the link is already updated in the pre-class.

There is the `./class_plan.md` file with the rough plan for the presentation.

Do you still think we should remove it?

Reply:
I think you can have a link for the pre-class installation instructions, I don;t think we need to have a detailed installation guide in the class material..
Let's see what is Caio's opinion on that.